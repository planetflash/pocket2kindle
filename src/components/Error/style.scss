.error {
  color: map-get($colors, error);
  font-weight: 700;
}
