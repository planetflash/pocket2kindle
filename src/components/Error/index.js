import React from "react";
import PropTypes from "prop-types";

import Fade from "transitions/Fade";

import "./style.scss";

const Error = ({ message }) => {
  const showError = message ? true : false;

  return (
    <div className="error">
      <Fade in={showError}>{message}</Fade>
    </div>
  );
};

Error.propTypes = {
  message: PropTypes.string.isRequired
};

export default Error;
