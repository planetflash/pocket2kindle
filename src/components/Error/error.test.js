import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Error from "./";

describe("Error component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Error message={null} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("shows the message text when there is a message", () => {
    const testMessage = "test message";
    const wrapper = shallow(<Error message={testMessage} />);
    const text = wrapper.find(".message").text();

    expect(text).toBe(testMessage);
  });

  it("does not show the message text when there is a no message", () => {
    const wrapper = shallow(<Error message={null} />);
    const text = wrapper.find(".message").text();

    expect(text).toBe("");
  });
});
