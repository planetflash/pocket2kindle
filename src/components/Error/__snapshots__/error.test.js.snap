// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component matches the snapshot 1`] = `
<div
  className="src-components-Error-___style__error___2rnxz"
>
  <div
    style={
      Object {
        "display": "block",
        "opacity": 0,
        "transition": "opacity 300ms ease-in-out",
      }
    }
  >
    <span
      className="message"
    />
  </div>
</div>
`;
